.github/workflows/main.yml

```diff
@@ -17,8 +17,10 @@ jobs:
         run: pip install -r requirements.txt
       - name: check project styling
         run: pre-commit run --all-files
-      - name: setup asv machine
-        run: asv machine --yes
+      - name: run tests
+        run: python -m py.test
+      - name: write asv machine
+        run: python write_asv_machine.py
```

Comment:
This gets out of hand :slightly_smiling_face: You are using pytest to monkeypatch asv.

You could just create a wrapper script that would do that(use monkeypatch directly), there is no reason to involve pytest.

Reply:
Pytest is used to run test, it has nothing to do with execution of `write_asv_machine`.
I introduced pytest here, because its the first change that requires testing, in my opinion. Having said so, current setup is confusing, Ill create separate job for testing and styling check and make it requirement for starting benchmark run.